src/librustc/hir/map/mod.rs

```diff
@@ -1328,28 +1308,27 @@ impl<'a> print::State<'a> {
     }
 }
 
-fn node_id_to_string(map: &Map<'_>, id: NodeId, include_id: bool) -> String {
-    let id_str = format!(" (id={})", id);
+fn hir_id_to_string(map: &Map<'_>, id: HirId, include_id: bool) -> String {
+    let id_str = format!(" (hir_id={})", id);
     let id_str = if include_id { &id_str[..] } else { "" };
 
     let path_str = || {
         // This functionality is used for debugging, try to use TyCtxt to get
         // the user-friendly path, otherwise fall back to stringifying DefPath.
         crate::ty::tls::with_opt(|tcx| {
             if let Some(tcx) = tcx {
-                let def_id = map.local_def_id(id);
+                let def_id = map.local_def_id_from_hir_id(id);
                 tcx.def_path_str(def_id)
-            } else if let Some(path) = map.def_path_from_id(id) {
+            } else {
+                let path = map.def_path_from_hir_id(id);
                 path.data.into_iter().map(|elem| {
                     elem.data.to_string()
                 }).collect::<Vec<_>>().join("::")
-            } else {
-                String::from("<missing path>")
```

Comment:
Why is the else path removed here?

Reply:
It was removed under the assumption that `local_def_id` and `local_def_id_from_hir_id` do essentially the same, except that the latter one does not return an `Option`. Since it does not return an `Option`, the `if let` became obsolete and with it the `else` clause.